File: 0122 best-time-to-buy-and-sell-stock-ii/solution.rs
impl Solution {
    pub fn max_profit(p: Vec<i32>) -> i32 {
        let mut sum=0;
        for i in 0..p.len()-1 {
            if p[i]<p[i+1] {
                sum+=p[i+1]-p[i];
            }
        }
        sum
    }
}

File: 0122 best-time-to-buy-and-sell-stock-ii/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int sum = 0;
        for(int i=1;i<p.size();i++) {
            if(p[i]>p[i-1]) sum+=p[i]-p[i-1];
        }
        return sum;
    }
};
